crates/subspace-farmer/src/multi_farming.rs

```diff
@@ -99,18 +116,100 @@ impl MultiFarming {
                             plot.clone(),
                             plot_commitments.clone(),
                             client.clone(),
-                            identity,
+                            identity.clone(),
                             reward_address,
                         )
                     });
 
-                    Ok::<_, anyhow::Error>((plot, subspace_codec, plot_commitments, farming))
+                    Ok::<_, anyhow::Error>((
+                        identity,
+                        plot,
+                        subspace_codec,
+                        plot_commitments,
+                        farming,
+                    ))
+                })
+            })
+            .collect::<FuturesOrdered<_>>()
+            .enumerate();
+
+        let starting_port = if let Some(mut starting_multiaddr) = listen_on.clone() {
+            if let Some(Protocol::Tcp(starting_port)) = starting_multiaddr.pop() {
+                Some(starting_port)
+            } else {
+                return Err(anyhow::anyhow!("Unknown protocol {}", listen_on.unwrap()));
+            }
+        } else {
+            None
+        };
+
+        while let Some((i, result)) = results.next().await {
+            let (identity, plot, subspace_codec, plot_commitments, farming) = result.unwrap()?;
```

Comment:
Could we have `expect()` instead of `unwrap()` to get an error message? I would suggest changing all unwraps to expect() in the new code.

Reply:
Those tasks shouldn't fail at all, I believe, so it should be fine just an unwrap as it is.

Reply:
We currently do not accept `unwrap` at all in blockchain client code, if you expect it to never happen here then just put `.expect()` with solid explanation why invariant is protected and panic will never happen.